java-data/src/main/java/net/cryptic_game/backend/data/Chat/Message.java

```diff
@@ -0,0 +1,147 @@
+package net.cryptic_game.backend.data.Chat;
+
+import com.google.gson.JsonObject;
+import net.cryptic_game.backend.base.sql.models.TableModelAutoId;
+import net.cryptic_game.backend.base.utils.JsonBuilder;
+import net.cryptic_game.backend.data.user.User;
+import org.hibernate.Session;
+import org.hibernate.annotations.Type;
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.Objects;
+
+@Entity
+@Table(name = "message")
+public class Message extends TableModelAutoId {
+
+    @ManyToOne
+    @JoinColumn(name = "user_id", updatable = false, nullable = false)
+    @Type(type = "uuid-char")
+    private User user;
+
+    @ManyToOne
+    @JoinColumn(name = "channel_id", nullable = false, updatable = false)
+    @Type(type = "uuid-char")
+    private Channel channel;
+
+    @Enumerated(EnumType.STRING)
+    @Column(updatable = false)
+    private ChatAction type;
+
+    @Column(name = "text", updatable = false, nullable = true)
+    private String text;
+
+    /**
+     * user id of the target, if it's not a whisper its null and targets consequently
+     * everyone in the channel
+     */
```

Comment:
Change to: User ID of the target, if it is not a whisper it is null and consequently targets everyone in the channel